libs/clj-icat-direct/src/clj_icat_direct/queries.clj

```diff
@@ -533,17 +533,16 @@
                             FROM r_data_main
                            WHERE coll_id = ANY(ARRAY( SELECT coll_id FROM parent )))
 
-    SELECT count(*) AS total
-      FROM ( SELECT DISTINCT d.data_id FROM r_objt_access a
+    SELECT ((SELECT count(DISTINCT d.data_id) FROM r_objt_access a
                JOIN data_objs d ON a.object_id = d.data_id
               WHERE a.user_id IN ( SELECT group_user_id FROM user_groups )
-                AND a.object_id IN ( SELECT data_id from data_objs )
-              UNION
-             SELECT DISTINCT c.coll_id FROM r_coll_main c
+                AND a.object_id IN ( SELECT data_id from data_objs ))
+            +
```

Comment:
I like using + here rather than the union; in my own playing with this I'd changed it to union all to remove the extra unique step, but this is even cleaner.